test/utils/IntegrationTest.sol

```diff
@@ -0,0 +1,415 @@
+        // Get protocol state information used for calculating correct LP shares.
+        (uint256 totalBase, uint256 totalShare, ) = getProtocolSharePrice();
+
+        // Contribution in terms of shares.
+        uint256 contributionShares = contribution.mulDivDown(
+            totalShare,
+            totalBase
+        );
+
+        // Deploy all Hyperdrive targets using deployer coordinator contract.
+        // This function reverts if base deposits are disabled.
+        deployTargets(
+            bytes32(uint256(0xbeefbabe)), // Deployment Id
+            bytes32(uint256(0xdeadfade)), // Deployment Salt
+            contribution, // Contribution
+            true // asBase
+        );
+
+        // If base deposits are enabled we verify some assertions.
+        if (config.enableBaseDeposits && isBaseETH) {
+            // If the base token is ETH we assert the ETH balance is correct.
+            if (isBaseETH) {
+                assertEq(
+                    address(alice).balance,
+                    aliceBalanceBefore - contribution
+                );
+            }
+
+            // Ensure that the decimals are set correctly.
+            assertEq(hyperdrive.decimals(), 18);
+
+            // Ensure that Alice received the correct amount of LP tokens. She should
+            // receive LP shares totaling the amount of shares that he contributed
+            // minus the shares set aside for the minimum share reserves and the
+            // zero address's initial LP contribution.
+            assertApproxEqAbs(
+                hyperdrive.balanceOf(AssetId._LP_ASSET_ID, alice),
+                contribution.divDown(
+                    hyperdrive.getPoolConfig().initialVaultSharePrice
+                ) - 2 * hyperdrive.getPoolConfig().minimumShareReserves,
+                config.shareTolerance // Custom share tolerance per integration.
+            );
+
+            // Ensure that the share reserves and LP total supply are equal and correct.
+            assertApproxEqAbs(
+                hyperdrive.getPoolInfo().shareReserves,
+                contributionShares,
+                1
+            );
+            assertEq(
+                hyperdrive.getPoolInfo().lpTotalSupply,
+                hyperdrive.getPoolInfo().shareReserves -
+                    hyperdrive.getPoolConfig().minimumShareReserves
+            );
+
+            // Verify that the correct events were emitted.
+            verifyFactoryEvents(
+                deployerCoordinator,
+                hyperdrive,
+                alice,
+                contribution,
+                FIXED_RATE,
+                true,
+                hyperdrive.getPoolConfig().minimumShareReserves,
+                new bytes(0),
+                config.shareTolerance
+            );
+        }
+    }
+
+    /// @dev Test to verify a market can be deployed and initialized funded
+    ///      by the share token.
+    function test__deployAndInitialize__asShares() external {
```

Comment:
For the sake of completeness, I'm wondering if we should have a config option about whether or not shares are supported. Right now, all of our yield sources support shares. That said, this may not always be the case, and it would be easier to implement this now than down the road.